req.go

```diff
@@ -245,38 +242,32 @@ func (r *Req) IsDeleted() bool {
 	return strings.HasPrefix(r.Title, "DELETED")
 }
 
-// checkAttributes validates the requirement attributes against the provided schema, returns a list of errors found
+// checkAttributes validates the requirement attributes against the schema from its document,
+// returns a list of errors found.
 // @llr REQ-TRAQ-SWL-10, REQ-TRAQ-SWL-29
-func (r *Req) checkAttributes(as Schema) []error {
+func (r *Req) checkAttributes() []error {
 	var errs []error
 	var anyAttributes []string
 	anyCount := 0
 
 	// Iterate the attribute rules
-	for name, rules := range as.Attributes {
-		for _, rule := range rules {
+	for name, attribute := range (*r.Document).Schema.Attributes {
```

Comment:
I think you can just write `for name, attribute := range r.Document.Schema.Attributes {`

Reply:
good point. I'm still unsure when the dereference needs to be explicit.